src/reader/components/NamedEntitiesModeReader.vue

```diff
@@ -65,17 +65,31 @@
       onShowMap(kind) {
         this.mapState = kind;
       },
+      getTokenLookup(entities) {
+        // @@@ likely this can be improved via map reduce
+        const tokenLookup = {};
+        entities.forEach(entity => {
+          const tokenVeRefs = entity.tokens.edges.map(t => t.node.veRef);
+          tokenVeRefs.forEach(veRef => {
+            tokenLookup[veRef] = tokenLookup[veRef] || [];
+            tokenLookup[veRef].push(entity.id);
+          });
+        });
+        return tokenLookup;
```

Comment:
Here is the map reduce version but I'm not sure that's any better than using forEach to build things up.

```suggestion
        return entities.reduce((map, entity) => {
          const veRefs = entity.tokens.edges.map(t => t.node.veRef).reduce((vmap, veRef) => {
            return {
              ...vmap,
              [veRef]: [...[vmap[vref] || []], entity.id],
            }
          }, {});
          return {
            ...map,
            ...veRefs,
          };
        }, {});
```

Reply:
There isn't anything requiring this to be a method on the component so if there is a change we'd reuse this it could stand on it's own as a function that could be shared.

Reply:
Thanks @paltman; the map reduce seems less readable to me, so I've kept the original implementation